runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -96,6 +96,7 @@ public void gc(long trimMark) {
      */
     public List<SMREntry> current() {
         ILogData data = streamView.current();
+
```

Comment:
@lixinchengdu I think your initial argument that the object layer should always see `SMREntry` makes more sense. But even on master, it seems like an empty list is returned (look at line 100). 

I think returning an empty list is error-prone can hide issues, maybe we should explicitly fail if the entry is not `ISMRConsumable` or a `DataType.DATA`